internal/proxy/powerscale_handler.go

```diff
@@ -0,0 +1,454 @@
+		}
+	})
+	if err != nil {
+		log.Printf("opa: %v", err)
+		h.writeError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	var resp struct {
+		Result struct {
+			Allow bool `json:"allow"`
+		} `json:"result"`
+	}
+	err = json.NewDecoder(bytes.NewReader(ans)).Decode(&resp)
+	if err != nil {
+		log.Printf("decode json: %q: %v", string(ans), err)
+		h.writeError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if !resp.Result.Allow {
+		log.Println("Request denied")
+		h.writeError(w, "request denied for path", http.StatusNotFound)
+		return
+	}
+
+	mux.ServeHTTP(w, r)
+}
+
+func (h *PowerScaleHandler) spoofLoginRequest(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.SpanFromContext(r.Context()).Tracer().Start(r.Context(), "spoofLoginRequest")
+	defer span.End()
+	_, err := w.Write([]byte("hellofromkaravi"))
+	if err != nil {
+		h.log.Printf("failed to write response: %v", err)
+	}
+}
+
+func (h *PowerScaleHandler) writeError(w http.ResponseWriter, msg string, code int) {
+	h.log.Printf("proxy: powerscale_handler: writing error:  %d: %s", code, msg)
+	w.WriteHeader(code)
+	errBody := struct {
+		Code       int    `json:"errorCode"`
+		StatusCode int    `json:"httpStatusCode"`
+		Message    string `json:"message"`
+	}{
+		Code:       code,
+		StatusCode: code,
+		Message:    msg,
+	}
+	err := json.NewEncoder(w).Encode(&errBody)
+	if err != nil {
+		log.Println("Failed to encode error response", err)
+		http.Error(w, "Failed to encode error response", http.StatusInternalServerError)
+	}
+}
+
+func (s *PowerScaleSystem) volumeCreateHandler(next http.Handler, enf *quota.RedisEnforcement, opaHost string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx, span := trace.SpanFromContext(r.Context()).Tracer().Start(r.Context(), "volumeCreateHandler")
+		defer span.End()
+
+		var systemID string
+		if v := r.Context().Value(web.SystemIDKey); v != nil {
+			var ok bool
+			if systemID, ok = v.(string); !ok {
+				//TODO(aaron): update writeError for powerscale
+				writeError(w, "powerscale", http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+				return
+			}
+		}
+
+		isiPath := strings.TrimPrefix(filepath.Dir(r.URL.Path), "/namespace")
+
+		// Read the body.
+		// The body is nil but we use the resulting io.ReadCloser to reset the request later on.
+		b, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			writeError(w, "powerscale", "failed to read body", http.StatusInternalServerError)
+			return
+		}
+		defer r.Body.Close()
+
+		// Get the remote host address.
+		host, _, err := net.SplitHostPort(r.RemoteAddr)
+		if err != nil {
+			writeError(w, "powerscale", "failed to parse remote host", http.StatusInternalServerError)
+			return
+		}
+		s.log.Printf("RemoteAddr: %s", host)
+
+		pvName := r.Header.Get(HeaderPVName)
+		log.Printf("pvName: %s", pvName)
+		// Update metrics counter for volumes requested.
+		//volReqCount.Add(pvName, 1)
```

Comment:
Remove this comment and commented code if not required

Reply:
Removed.